zkvm/src/vm.rs

```diff
@@ -51,6 +80,96 @@ pub struct Tx {
     pub proof: R1CSProof,
 }
 
+impl Tx {
+    fn encode(&self, buf: &mut Vec<u8>) {
+        self.header.encode(buf);
+        encoding::write_size(self.program.len(), buf);
+        buf.extend(&self.program);
+        buf.extend_from_slice(&self.signature.to_bytes());
+        buf.extend_from_slice(&self.proof.to_bytes());
+    }
+
+    fn decode<'a>(reader: &mut SliceReader<'a>) -> Result<Tx, VMError> {
+        let (tx, _) = reader.slice(|r| {
```

Comment:
as in the above comment, no need for `.slice` wrapper, since we do not want to use that slice in any way.